metadata/images.go

```diff
@@ -55,6 +55,37 @@ func (s *imageStore) Get(ctx context.Context, name string) (images.Image, error)
 
 	if err := view(ctx, s.db, func(tx *bolt.Tx) error {
 		bkt := getImagesBucket(tx, namespace)
+		if bkt == nil {
+			nsbkt := getNamespacesBucket(tx)
```

Comment:
How about extracting this block as a function? It is a bit lengthy and deeply-nested.

Reply:
+1 on extracting.  It'd also be nice to see a unit test (or several) that covers the new functionality here.